Libraries/Components/Touchable/TouchableOpacity.js

```diff
@@ -102,20 +106,45 @@ var TouchableOpacity = React.createClass({
    * defined on your component.
    */
   touchableHandleActivePressIn: function() {
-    this.refs[CHILD_REF].setNativeProps({
-      opacity: this.props.activeOpacity
-    });
+    this._fromPressIn = true;
+    this.clearTimeout(this._hideTimeout);
+    this._hideTimeout = null;
+    this._opacityActive();
     this.props.onPressIn && this.props.onPressIn();
   },
 
   touchableHandleActivePressOut: function() {
-    var child = onlyChild(this.props.children);
-    var childStyle = flattenStyle(child.props.style) || {};
-    this.setOpacityTo(childStyle.opacity === undefined ? 1 : childStyle.opacity);
-    this.props.onPressOut && this.props.onPressOut();
+    if (this.props.delayPressOut) {
+      this._onPressOutTimeout = this.setTimeout(function() {
+        this._opacityInactive();
+        this.props.onPressOut && this.props.onPressOut();
+      }, this.props.delayPressOut);
+    } else {
+      this._opacityInactive();
+      this.props.onPressOut && this.props.onPressOut();
+    }
   },
 
   touchableHandlePress: function() {
+    if (this.props.delayPress) {
+      if (!this._onPressTimeout) {
+        this._onPressTimeout = this.setTimeout(function() {
+          this.clearTimeout(this._onPressTimeout);
+          this._onPressTimeout = null;
+          this._onPress();
+        }, this.props.delayPress);
```

Comment:
Same with touchableHandleActivePressOut.  And I don't mean completely share the impl, but can you factor out the common logic into Touchable.js?